### ISSUE TITLE
Is "skip_serializing_if" not supported yet?

### ISSUE THREAD

**Issue by mlevkov** [media-io/yaserde#43]
Hello,

I've various attributes and structs that are optionally present based on the context. 
I see that "skip_serializing_if" is not checked as supported. Before I consider rewriting my codebase to use your code, would you be kind to let me know if "skip_serializing_if" is supported? 

Highest Regards,
Max

**Comment by MarcAntoine-Arnaud** [comment]
No yet supported, it's a nice feature ;-)

**Comment by mlevkov** [comment]
I'm not that verse in Rust yet, otherwise, I'd love to contribute :). The library looks really amazing. Is getting "skip_serializing_if" supported a fairly complicated task? I'm particularly interested in "Option::is_none" case, not sure what the other use cases look like. Is there is a way to work around the issue that you can recommend in the meanwhile? Would you think that I can either pair your library with Serde-rs use or I should completely replace it with yaserde?

**Comment by mlevkov** [comment]
I just ported the code and it appears that the library is "automatically" skips the `None` cases, which is nice :). However, I've discovered that output is now including the escape characters. Would it be kind to recommend a way to signal not to include escape characters, such as newline or quotes, or it would be better to do so in post-processing?

**Comment by MarcAntoine-Arnaud** [comment]
Hi @mlevkov 
The `skip_serializing_if` can be awesome if it's generic. `Option::is_none` is a great use case, but it can be great if a generic value will not be serialized if it's not needed (like a 0 or empty string if it's a default value).
For that, I like the idea of passing in parameter a function pointer which can take `self` and return bool.
Like that it can be generic.
It's not very complex, it can be a great first task I think.

Regarding serde-rs, I have asked 2 years ago to improve the XML mapping. They don't wanna add complexity to support specific de-serializing with XML specification (namespace and attributes).
That's why I have started this project and still independant of serde.
But YaSerDe can be use simultaneously with SerDe on strutures/enums. So you can serialise in json and xml the same model !

**Comment by mlevkov** [comment]
Hi @MarcAntoine-Arnaud,

If I wanted to contribute by developing this "generic" feature, what kind of things do I need to look at? Not only that I'm learning Rust ;), but I've never worked with details pertaining to the development of attributes, aside from using them within the library. Nonetheless, I've noticed that serialization skips none values already, isn't that so?

Yes, I agree properly composed data structure can act in many ways,  including passing details as JSON or another similar format, that is what I'm after here. I've many different microservices that are moving data among each other, within my architecture, I'd keen on evolving a model. 

I've landed here because of the mention in the threads of the serde-rs.
